### PR TITLE
DOC-524: Document Redis collection data types

### PR DESCRIPTION
**Pull request by diana-macrometa**
Based on [DB-1830](https://macrometa.atlassian.net/browse/DB-1830), document data format requirements for Redis collections.

### CODE REVIEW

File: docs/collections/redis-mode/redis-data-format.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+---
+title: Redis Data Format
+sidebar_position: 50
+---
+
+Redis Mode collections require data to be stored in certain formats in order to be compatible with Redis.
+
+If you add items to the collection using the Redis API, then the format is enforced automatically. However, if you add items through Macrometa, then you must ensure that data follows the following formats according to type.
```
Comment (martin-macrometa) [inline docs/collections/redis-mode/redis-data-format.md]: Redis API is still part of macrometa. I would phrase the second sentence a little differently:

However, if you add items through other APIs, then you must ensure that data respects the following formats according to type.

File: docs/collections/redis-mode/redis-data-format.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+## List
+
+```json
+{key: ..., list: [...], data_type: "list"}
+```
+
+## Hash
+
+This example shows multiple entries per hash.
```
Comment (martin-macrometa) [inline docs/collections/redis-mode/redis-data-format.md]: "This example shows" -> "There can be"

File: docs/collections/redis-mode/redis-data-format.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+```json
+{key: ..., value: ..., hvalue: ..., data_type: "hash"}
+```
+
+## Set
+
+This example shows multiple entries per set.
```
Comment (martin-macrometa) [inline docs/collections/redis-mode/redis-data-format.md]: "This example shows" -> "There can be"

File: docs/collections/redis-mode/redis-data-format.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+```json
+{key: ..., value: ..., data_type: "set"}
+```
+
+## Sorted Set
+
+This example shows multiple entries per sorted set.
```
Comment (martin-macrometa) [inline docs/collections/redis-mode/redis-data-format.md]: "This example shows" -> "There can be"

File: docs/collections/redis-mode/redis-data-format.md (hunk 1 above)
Comment (diana-macrometa) [reply docs/collections/redis-mode/redis-data-format.md]: FYI, if you click the little page icon with the plus and minus, then you can add a suggestion directly.

**Review by martin-macrometa — APPROVED** [review]

**Pull request merged**